plugins/modules/find_perl_module.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/python
+
+# Copyright (c) 2023, Jonathan Kamens <jik@kamens.us>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# pylint: disable=missing-module-docstring
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: find_perl_module
+
+short_description: Determine from where to install Perl modules
+
+requirements:
+    - C(apt-file) executable in search path on systems that use the apt package
+      manager
+    - C(dnf) or C(yum) executable in search path on systems that use the
+      dnf/yum package manager
+    - C(cpanm) executable in search path if you want to be able to search for
+      packages using cpanm
+
+version_added: "1.0.0"
+
+description:
+    - Searches dnf, yum, apt, and/or cpanm to determine the best source from
+      which to install a Perl module.
+    - Prefers the OS repositories over cpanm.
+    - Specify module names as you would specify them with the C(use) command in
+      a Perl script.
+    - Does not actually install modules. Instead, returns information about
+      where they can be installed from, which can be supplied to subsequent
+      tasks to do the actual installation.
+    - Note that this module will not fail by default if it cannot locate a
+      requested module. If you want that behavior, include a C(failed_when)
+      which checks for C(missing) being non-empty.
+
+options:
+    name:
+        description: One or more modules to search for
+        required: true
+        type: list
+        elements: str
+    try_installed:
+        description: Whether to check if modules are already installed and not
+          look elsewhere if they are
+        type: bool
+        default: true
+    try_dnf:
+        description: Whether to check for modules using the dnf package manager
+        default: true if dnf executable is available
+    try_yum:
+        description: Whether to check for modules using the dnf package manager
+        default: true if C(try_dnf) is false and yum executable is available
+    try_apt:
+        description: Whether to check for modules using the apt package manager
+        default: true if apt-file executable is available
+    try_cpanm:
+        description: Whether to check for modules using cpanm
+        default: true if cpanm executable is available
+    update:
+        description: Whether to update package manager databases before
+          searching
+        type: bool
+        default: false
+
```

Comment:
Please add
```
extends_documentation_fragment:
    - community.general.attributes
attributes:
    check_mode:
        support: ???
    diff_mode:
        support: ???
```
somewhere with the support values filled out correctly. (If this module is named `_info` this needs to look differently - look at other `_info` modules.)